classes/profile.php

```diff
@@ -16,35 +16,41 @@
 
 namespace tool_excimer;
 
-defined('MOODLE_INTERNAL') || die();
-
 /**
- * Functions relevant to profiles.
+ * Profile saving/loading and manipulation.
  *
  * @package   tool_excimer
- * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>, Kevin Pham <kevinpham@catalyst-au.net>
  * @copyright 2021, Catalyst IT
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class profile {
+    const TABLE = 'tool_excimer_profiles';
+
 
-    /** Request's fallback value for when the $SCRIPT is null */
-    const REQUEST_UNKNOWN = 'UNKNOWN';
+    /** Reason - MANUAL - Profiles are manually stored for the request using FLAMEME as a page param. */
+    const REASON_MANUAL   = 0b0001;
```

Comment:
kinda off topic but it also seems inconsistent now in hindsight that we have REASON_FLAMEALL vs REASON_MANUAL. I think the latter should be REASON_FLAMEME

Reply:
done